dsp-repository/v1/dsp-repository.shacl.ttl

```diff
@@ -26,32 +26,670 @@
+# dsp-repo:hasBooBoo
+#     a rdf:Property .
+
+##### CLASS DEFINITIONS #####
+### --- Class definition of dsp-repo:Project --- 
+# http://ns.dasch.swiss/repository#Project
+dsp-repo:Project 
+    a rdfs:Class ;
+    rdfs:label "dsp-repo:Project"@en ;
+    rdfs:comment "A project"@en ;
+    rdfs:subClassOf schema:Project , prov:Entity .   
+
+### --- Class definition of dsp-repo:Dataset --- 
+# http://ns.dasch.swiss/repository#Dataset
+dsp-repo:Dataset 
+    a rdfs:Class ;
+    rdfs:comment "A dataset"@en ;
+    rdfs:label "dsp-repo:Dataset"@en ;
+    rdfs:subClassOf schema:Dataset .
+
+### --- Class definition of dsp-repo:Person --- 
+# http://ns.dasch.swiss/repository#Person
+dsp-repo:Person 
+    a rdfs:Class ;
+    rdfs:comment "A person"@en ;
+    rdfs:label "dsp-repo:Person"@en ;
+    rdfs:subClassOf schema:Person , prov:Agent . 
+
+### --- Class definition of dsp-repo:Organization --- 
+# http://ns.dasch.swiss/repository#Organization
+dsp-repo:Organization 
+    a rdfs:Class ;
+    rdfs:comment "An organization"@en ;
+    rdfs:label "dsp-repo:Organization"@en ;
+    rdfs:subClassOf schema:Organization , prov:Agent . 
+
+### --- Class definition of dsp-repo:Grant --- 
+# http://ns.dasch.swiss/repository#Grant
+dsp-repo:Grant 
+    a rdfs:Class ;
+    rdfs:comment "A grant"@en ;
+    rdfs:label "dsp-repo:Grant"@en ;
+    rdfs:subClassOf schema:Grant . 
+
+### --- Class definition of dsp-repo:DataManagementPlan --- 
 # http://ns.dasch.swiss/repository#DataManagementPlan
 dsp-repo:DataManagementPlan
-    a owl:Class, sh:NodeShape ;
-    rdfs:label "dsp-repo:DataManagementPlan"@en ;
+    a rdfs:Class ;
     rdfs:comment "A Data Management Plan"@en ;
-    rdfs:subClassOf schema:Thing ;
+    rdfs:label "dsp-repo:DataManagementPlan"@en ;
+    rdfs:subClassOf schema:Thing .
+
+##### SHAPES #####
+### --- Shape of dsp-repo:Project ---
+dsp-repo:ProjectShape
+    a sh:NodeShape ;
+    sh:targetClass dsp-repo:Project ;
+    sh:property [
+        sh:path dsp-repo:hasName ;
+        sh:name "Name" ;
+        sh:description "Name of the project"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+        sh:minCount 1 ;
+    ] ; 
+    sh:property [
+        sh:path dsp-repo:hasDescription ;
+        sh:name "Description" ;
+        sh:description "Description of the project"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+        sh:minCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasKeywords ;
+        sh:name "Keywords" ;
+        sh:description "Keywords and tags"@en ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasDiscipline ;
+        sh:name "Discipline" ;
+        sh:description "Discipline and research fields from UNESCO nomenclature: https://skos.um.es/unesco6/?l=en or from http://www.snf.ch/SiteCollectionDocuments/allg_disziplinenliste.pdf"@en ;
+        sh:or ( [ sh:datatype xsd:string ] [ sh:class schema:URL ] ) ;
```

Comment:
Since I don't understand SHACL yet, why 
```ttl
[ sh:class schema:URL ]
```
and not
```ttl
[ sh:nodeKind sh:IRI  ]
```
? Do we have to rely on schema.org on this one?

Reply:
I'm new to SHACL too, so don't hold me to it... But if I got that right, then `sh:nodeKind sh:IRI` only defines what kind of Node there is at this point, and in SHACL Nodes can be either IRIs, Blanknodes or Literals.

So, "http://www.example.org" would be a literal (and thus not match `sh:IRI`), whereas having an IRI pointing e.g. to a `dsp-repo:Person` would be valid, although it clearly isn't a URL.

@danielasubotic correct me, if I'm mistaken.

Reply:
Please try to use the example in the metadata example:
`<test-project> dsp-repo:hasDiscipline [
   a schema:URL ;
   schema:propertyID [
       a schema:PropertyValue ;
       schema:propertyID "SKOS UNESCO Nomenclature" ;
   ] ;
   schema:url "http://skos.um.es/unesco6/11" ;
] .`

Reply:
You are right, it works perfectly. I guess I spent too long a time with OWL and away from schema.org but I am missing something in this logic. Maybe you can explain it to me tomorrow?

```ttl
<test-project> dsp-repo:hasDiscipline [ 
a schema:URL ; 
schema:propertyID [ 
a schema:PropertyValue ; 
schema:propertyID "SKOS UNESCO Nomenclature" ; ] ; 
schema:url "http://skos.um.es/unesco6/11" ; ] .
```
I don't understand:
- why we need to declare in the data file
```ttl 
<test-project> dsp-repo:hasDiscipline [ 
a schema:URL ]; 
```
- and also if we need to declare it, then why don't we do it in the ontology instead?